### PR TITLE
DCS-863 repackaging services and routes

### PR DESCRIPTION
**Pull request by andrewrlee**
Little bit of refactoring of the services:

*  `services/reports` -> `services/drafts`  as its all around creating and managing draft reports.
* extracted out a `services/reporting``

Refactoring routes:

We used to have one big routes folder and one large routes registration files
splitting out the really large routes folder into 4 sections each exposing an index.ts file that returns a router (much like BVL):

* creatingReports 
* maintainReports
* viewingReports
* api

Names up for discussion,

### CODE REVIEW

File: server/app.ts [hunk 1]
```diff
@@ -250,8 +249,7 @@ export default function createApp(services: Services): Express {
 
   app.use(authorisationMiddleware)
 
-  app.use('/', createRouter(authenticationMiddleware, services))
-  app.use('/api/', createApiRouter(authenticationMiddleware, services))
```
Comment (andrewrlee) [inline server/app.ts:254]: move api into createRouter which simplifies the testing approach

File: server/routes/api.test.js [hunk 2]
```diff
@@ -14,35 +13,30 @@ const reportingService = {
...
 
-  describe('GET /offender/:bookingId/image', () => {
+  describe('GET /api/offender/:bookingId/image', () => {
     beforeEach(() => {
       userSupplier.mockReturnValue(user)
     })
     it('should render using system creds for retrieving image', async () => {
       await request(app)
-        .get('/offender/1234/image')
+        .get('/api/offender/1234/image')
```
Comment (andrewrlee) [inline server/routes/api.test.js:39]: now the route is hosted on the actual path that is used in production.

File: server/routes/creatingReports/addInvolvedStaff.test.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import request from 'supertest'
-import { StaffDetails } from '../data/draftReportClientTypes'
-import DraftReportService, { AddStaffResult } from '../services/report/draftReportService'
-import { appWithAllRoutes, user } from './testutils/appSetup'
+import { StaffDetails } from '../../data/draftReportClientTypes'
+import DraftReportService, { AddStaffResult } from '../../services/drafts/draftReportService'
+import { appWithAllRoutes, user } from '../__test/appSetup'
```
Comment (andrewrlee) [inline server/routes/creatingReports/addInvolvedStaff.test.ts:4]: test resources are along side normal code - making this more apparent by adopting this naming convention for the folder

File: server/routes/creatingReports/checkYourAnswers.ts [hunk 4]
```diff
@@ -65,4 +65,8 @@ export default class CheckAnswerRoutes {
     const location = reportId ? `/${reportId}/report-sent` : `/`
     return res.redirect(location)
   }
+
+  public viewReportSent = (req: Request, res: Response): void => {
+    return res.render('pages/report-sent', { data: res.locals.formObject, reportId: req.params.reportId })
+  }
```
Comment (andrewrlee) [inline server/routes/creatingReports/checkYourAnswers.ts:71]: It was really easy splitting the routes into these three areas which reflects that the division already existed to some extent in the code. 

This was the only route that had to be moved it, it was previously in the `incidents.ts` set of routes. Seemed to make sense that this moved to `creatingReports` as  this is the final step of creating a route - viewing the confirmation page

File: server/routes/index.ts [hunk 5]
```diff
@@ -160,9 +23,10 @@ export default function Index(
     next()
   })
 
-  userRoutes()
-  reviewerRoutes()
-  coordinatorRoutes()
+  router.use(creatingReportsRoutes(services))
+  router.use(maintainingReportsRoutes(services))
+  router.use(viewingReportsRoutes(services))
```
Comment (andrewrlee) [inline server/routes/index.ts:28]: much easier to see whats going on here

File: server/routes/maintainingReports/coordinator.ts [hunk 6]
```diff
@@ -0,0 +1,130 @@
+import type { InvolvedStaffService, OffenderService, ReportService, ReviewService } from '../../services'
+import type { SystemToken } from '../../types/uof'
+import { AddStaffResult } from '../../services/involvedStaffService'
+import { firstItem } from '../../utils/utils'
+import { ReportStatus } from '../../config/types'
+
+export default class CoordinatorRoutes {
```
Comment (andrewrlee) [inline server/routes/maintainingReports/coordinator.ts:7]: moved this from a js function to a typescript class

File: server/routes/signIn.js [hunk 7]
```diff
@@ -1,31 +0,0 @@
-const express = require('express')
```
Comment (andrewrlee) [inline server/routes/signIn.js:1]: deleted this bad boy as it wasn't used!

**Review by bell-pa — APPROVED** [review]
LGTM

**Pull request merged**